PhotonMapping/src/Mesh.cpp

```diff
@@ -3,45 +3,55 @@
 #include <iostream>
 
 Mesh::Mesh(std::vector<Vertex> vertices, std::vector<unsigned int> indices, RTCDevice device) {
-    this->vertices = vertices;
-    this->indices = indices;
-    
-    setupMesh(device);
+  this->vertices = vertices;
+  this->indices = indices;
+
+  _setupMesh(device);
+}
+
+std::pair<RTCGeometry, Material> Mesh::getMaterialPair() const {
+  return { _geometry, _material };
 }
 
-void Mesh::setupMesh(RTCDevice device) {
-    _geometry = rtcNewGeometry(device, RTC_GEOMETRY_TYPE_TRIANGLE);
-    vertexBuffer = (float*)rtcSetNewGeometryBuffer(_geometry,
-                                                   RTC_BUFFER_TYPE_VERTEX,
+void Mesh::_setupMesh(RTCDevice device) {
+  _geometry = rtcNewGeometry(device, RTC_GEOMETRY_TYPE_TRIANGLE);
+  vertexBuffer = (float*)rtcSetNewGeometryBuffer(_geometry,
+                                                 RTC_BUFFER_TYPE_VERTEX,
+                                                 0,
+                                                 RTC_FORMAT_FLOAT3,
+                                                 3 * sizeof(float),
+                                                 vertices.size());
+
+  std::cout << "size:" << vertices.size() << std::endl;
+
+  indexBuffer = (unsigned*)rtcSetNewGeometryBuffer(_geometry,
+                                                   RTC_BUFFER_TYPE_INDEX,
                                                    0,
-                                                   RTC_FORMAT_FLOAT3,
-                                                   3 * sizeof(float),
-                                                   vertices.size());
-    
-    std::cout << "size:" << vertices.size() << std::endl;
-    
-    indexBuffer = (unsigned*)rtcSetNewGeometryBuffer(_geometry,
-                                                     RTC_BUFFER_TYPE_INDEX,
-                                                     0,
-                                                     RTC_FORMAT_UINT3,
-                                                     1 * sizeof(unsigned),
-                                                     indices.size());
-    
-    for (unsigned int i = 0; i < vertices.size(); ++i) {
-        vertexBuffer[i * 3] = vertices[i].Position.x;
-        vertexBuffer[i * 3 + 1] = vertices[i].Position.y;
-        vertexBuffer[i * 3 + 2] = vertices[i].Position.z + 2;
-    }
-
-    for (unsigned int i = 0; i < indices.size(); ++i) {
-        indexBuffer[i] = indices[i];
-    }
+                                                   RTC_FORMAT_UINT3,
+                                                   1 * sizeof(unsigned),
+                                                   indices.size());
+
+  for (unsigned int i = 0; i < vertices.size(); ++i) {
+    vertexBuffer[i * 3] = vertices[i].Position.x;
+    vertexBuffer[i * 3 + 1] = vertices[i].Position.y;
+    vertexBuffer[i * 3 + 2] = vertices[i].Position.z + 2;
```

Comment:
Why `position.z + 2`?

Reply:
We changed this to move the object up a little bit since we don't have a Transfrom yet